### PR TITLE
PR for: prune away some of the annotations before the release that are not useful for users #4346

### PR DESCRIPTION
**Pull request by joeflack4**
## Updates
```
- Create: src/ontology/config/: Created this directory and initialized with a config file to be used by new makefile command for special annotations removal.
- Update: Updated makefile: New command: filtered2.owl: Extends filtered.owl and runs an annotation removal command. ```

### CODE REVIEW

**Comment by joeflack4** [comment]
(still working on this; i will attach some files w/ diffs)

File: src/ontology/config/remove-annotations-before-release.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+http://purl.obolibrary.org/obo/mondo#CLINGEN_PREFERRED
+http://purl.obolibrary.org/obo/mondo#clingen_preferred
+http://purl.obolibrary.org/obo/mondo#excluded_from_qc_check
+http://purl.obolibrary.org/obo/mondo#confidence
+MONDO:preferredExternal
```
Comment (matentzn) [inline src/ontology/config/remove-annotations-before-release.txt:5]: The last step will never work (removing "preferredExternal"

**Review by matentzn — CHANGES_REQUESTED** [review]
Looks great. Can you please provide a diff of filtered.owl owl with and without your change? any diff is fine, including command line, but robot diff is preferred.

**Comment by matentzn** [comment]
This PR works for "excluded_from_qc_check" and "confidence", but not for the other two things. We need to add SPARQL DELETE commands for that, lets discuss at call.

**Comment by nicolevasilevsky** [comment]
Should I merge this in so the two things are fixed for the Feb release (tomorrow) or should I hold off?
@matentzn

**Comment by matentzn** [comment]
This does not work yet.. Sorry. We need to push the Mondo release a few days, maybe to Friday.

**Comment by nicolevasilevsky** [comment]
sure, can do

File: src/ontology/config/remove-annotations-before-release.txt (hunk 1 above)
Comment (joeflack4) [reply src/ontology/config/remove-annotations-before-release.txt:5]: roger that

**Comment by joeflack4** [comment]
It sucks that my internet was down last Friday. I was hoping to discuss this PR at that time.

_Nico wrote:_
> Looks great. Can you please provide a diff of filtered.owl owl with and without your change? any diff is fine, including command line, but robot diff is preferred.

@matentzn : You can find the diff at [the bottom of my length report in the issue thread](https://github.com/monarch-initiative/mondo/issues/4346#issuecomment-1020776892).

_Nico wrote:_
> This PR works for "excluded_from_qc_check" and "confidence", but not for the other two things. We need to add SPARQL DELETE commands for that, lets discuss at call.

@matentzn Darn. Do you think that is a bug in `robot remove`? Perhaps something that would need to be updated in `robot`? If so, I can create a new GitHub issue for that if you like. But in any case, it sounds like I need to update my makefile recipe to somehow run a SPARQL query after the 'robot remove' bit. I think maybe you can instruct me on the best way to do this.

**Comment by matentzn** [comment]
@joeflack4 Yeah I found your diff on the issue later, thank you!

> Do you think that is a bug in robot remove? Perhaps something that would need to be updated in robot? If so, I can create a new GitHub issue for that if you like. But in any case, it sounds like I need to update my makefile recipe to somehow run a SPARQL query after the 'robot remove' bit. I think maybe you can instruct me on the best way to do this.

This is not strictly speaking a problem with ROBOT remove. Its a missing feature which I requested a long while back: https://github.com/ontodev/robot/issues/886.

The way you solve it is using an update query like this:

https://github.com/monarch-initiative/mondo/blob/master/src/sparql/update/rm-related-exact-synonym-annotations.ru

As we have to make a release on Friday, we need to get this done beforehand. I can do it, but I do not want to deprive you of the learning experience. Do you want to:

- [ ] Add an update query like the one linked above that specifically selects axiom annotations involving "http://purl.obolibrary.org/obo/mondo#clingen_preferred and "MONDO:preferredExternal".
- [ ] Add this query to src/sparql/update
- [ ] Run the following:

```
cd src/ontology
sh run.sh make update-merge-normalise-joes-update-query
```
- [ ] Push resulting changes to `mondo-edit.obo` to this branch

**Comment by joeflack4** [comment]
@matentzn Thanks for the explanation. I later on saw your comments on the diff, etc. I feel like I'm being pulled in a lot of directions this week (from within my team; not you). I'll try my best to stave off too many requests for my time and see if I can make this change on Thursday.

**Comment by matentzn** [comment]
@joeflack4 if you feel lile you are drowning, just ping me and I will do this for you so that @nicolevasilevsky can do her release. That is what teams are for :P You have done most of what is needed anyways.

**Comment by joeflack4** [comment]
@matentzn I do feel like I'm drowning this week; and I haven't even put any effort into Mondo/monarch stuff this week. Just a lot of demands from other grants this week. Sorry, and thanks!

**Comment by matentzn** [comment]
I have combined @joeflack4 solution now with a SPARQL delete query that will get rid of the MONDO:preferredExternal and CLINGEN_PREFERRED annotations. @nicolevasilevsky I did test it but I would take a note to make sure these are really removed in `mondo.owl` after the first step (`make all`) of the release.

**Review by matentzn — APPROVED** [review]

**Comment by matentzn** [comment]
Just dont let @cmungall see this! Quickly @nicolevasilevsky, quickly merge this!! Faster!

**Review by matentzn — APPROVED** [review]

**Review by nicolevasilevsky — APPROVED** [review]

**Comment by nicolevasilevsky** [comment]
Ok, I'm merging!

**Pull request merged**